js_modules/dagster-ui/packages/ui-core/src/assets/auto-materialization/GlobalAutomaterializationContent.tsx

```diff
@@ -0,0 +1,225 @@
+const FIVE_MINUTES = 5 * MINUTE;
+const TWENTY_MINUTES = 20 * MINUTE;
+
+export const GlobalAutomaterializationContent = () => {
+  useTrackPageView();
+
+  const automaterialize = useAutomaterializeDaemonStatus();
+  const confirm = useConfirmation();
+
+  const {permissions: {canToggleAutoMaterialize} = {}} = useUnscopedPermissions();
+
+  const [isPaused, setIsPaused] = useState(false);
+  const [statuses, setStatuses] = useState<undefined | InstigationTickStatus[]>(undefined);
+  const [timeRange, setTimerange] = useState<undefined | [number, number]>(undefined);
+  const getVariables = useCallback(
+    (now = Date.now()) => {
+      if (timeRange || statuses) {
+        return {
+          afterTimestamp: timeRange?.[0],
+          beforeTimestamp: timeRange?.[1],
+          statuses,
+        };
+      }
+      return {
+        afterTimestamp: (now - TWENTY_MINUTES) / 1000,
+      };
+    },
+    [statuses, timeRange],
+  );
+
+  const [fetch, queryResult] = useLazyQuery<AssetDaemonTicksQuery, AssetDaemonTicksQueryVariables>(
+    ASSET_DAEMON_TICKS_QUERY,
+  );
+
+  useLayoutEffect(() => {
+    fetch({variables: getVariables()});
+  }, [fetch, getVariables]);
+
+  const refresh = useCallback(
+    async () => await fetch({variables: getVariables()}),
+    [fetch, getVariables],
+  );
+
+  useRefreshAtInterval({
+    refresh,
+    enabled: !isPaused && !timeRange && !statuses,
+    intervalMs: 2 * 1000,
+  });
+
+  const [selectedTick, setSelectedTick] = useState<AssetDaemonTickFragment | null>(null);
+
+  const [tableView, setTableView] = useQueryPersistedState<'evaluations' | 'runs'>(
+    useMemo(
+      () => ({
+        queryKey: 'view',
+        decode: ({view}) => (view === 'runs' ? 'runs' : 'evaluations'),
+        encode: (raw) => {
+          return {view: raw, cursor: undefined, statuses: undefined};
+        },
+      }),
+      [],
+    ),
+  );
+
+  const data = queryResult.data ?? queryResult.previousData;
+
+  const allTicks = useMemo(() => {
+    return data?.autoMaterializeTicks || [];
+  }, [data]);
+
+  const ticks = useMemo(
+    () => {
+      return (
+        allTicks?.map((tick, index) => {
+          const nextTick = ticks[index - 1];
+          // For ticks that get stuck in "Started" state without an endTimestamp.
+          if (nextTick && isStuckStartedTick(tick, index)) {
+            const copy = {...tick};
+            copy.endTimestamp = nextTick.timestamp;
+            copy.status = InstigationTickStatus.FAILURE;
+            return copy;
+          }
+          return tick;
+        }) ?? []
+      );
+    },
+    // The allTicks array changes every 2 seconds because we query every 2 seconds.
+    // This would cause everything to re-render, to avoid that we memoize the ticks array that we pass around
+    // using the ID and status of the ticks.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [JSON.stringify(allTicks.map((tick) => `${tick.id}:${tick.status}`))],
```

Comment:
This makes sense and I think this is a good solution, but is it always the /last/ tick that changes in this array (eg: is allTicks append-only?)  Maybe something on line 92 could basically say "use the new data from the query response, but only if the last tick is not the last tick in state"

Reply:
@bengotow this is my logic he moved around and we need to take the status into account. So it's append only but the status of any item could change, not just the last item.